quickannotator/api/v1/tile/__init__.py

```diff
@@ -25,134 +24,125 @@ class PredictTileRespSchema(Schema):
+        if args['hasgt']:
+            ids = db_session.query(models.Tile.tile_id).filter(
+                models.Tile.annotation_class_id == annotation_class_id,
+                models.Tile.image_id == image_id,
+                models.Tile.tile_id.in_(ids),
+                models.Tile.gt_datetime.isnot(None)
+            ).all()
+            ids = [id[0] for id in ids]
+        return {"tileids": ids}, 200
+    
+@bp.route('/<int:image_id>/<int:annotation_class_id>/search/polygon')
+class TileIdSearchByPolygon(MethodView):
+    @bp.arguments(SearchTileByPolygonArgsSchema, location='json')
+    @bp.response(200, TileIdRespSchema)
+    def post(self, args, image_id, annotation_class_id):
+        """     get all Tiles within a polygon
+        """
+        tiles_in_polygon, _, _ = get_tile_ids_intersecting_polygons(image_id, annotation_class_id, [args['polygon']], mask_dilation=1)
+        tile_ids_in_mask, _, _ = get_tile_ids_intersecting_mask(image_id, annotation_class_id, mask_dilation=1)
+        ids = set(tiles_in_polygon) & set(tile_ids_in_mask)
+
+        if args['hasgt']:
+            ids = db_session.query(models.Tile.tile_id).filter(
+                models.Tile.annotation_class_id == annotation_class_id,
+                models.Tile.image_id == image_id,
+                models.Tile.tile_id.in_(ids),
+                models.Tile.gt_datetime.isnot(None)
+            ).all()
+            ids = [id[0] for id in ids]
+        return {"tileids": ids}, 200
+
     
-@bp.route('/search/coordinates')
-class TileSearchByCoordinates(MethodView):
+@bp.route('/<int:image_id>/<int:annotation_class_id>/search/coordinates')
+class TileIdSearchByCoordinates(MethodView):
     @bp.arguments(SearchTileByCoordinatesArgsSchema, location='query')
-    @bp.response(200, TileRespSchema)
-    def get(self, args):
+    @bp.response(200, TileIdRespSchema)
+    def get(self, args, image_id, annotation_class_id):
         """     get a Tile for a given point
         """
-        tilespace = get_tilespace(image_id=args['image_id'], annotation_class_id=args['annotation_class_id'], in_work_mag=False)
+        tilespace = get_tilespace(image_id=image_id, annotation_class_id=annotation_class_id, in_work_mag=False)
         tile_id = tilespace.point_to_tileid(args['x'], args['y'])
-        return get_tile(args['annotation_class_id'], args['image_id'], tile_id), 200
-
-@bp.route('/predict')
-class TilePredict(MethodView):
-    @bp.arguments(PredictTileArgsSchema, location='json')
-    @bp.response(201, PredictTileRespSchema)
-    def post(self, args):
-        """     predict tiles for a given image & class
-        """
-        upsert_tile(args['annotation_class_id'], args['image_id'], args['tile_id'], seen=TileStatus.PROCESSING)
-
-        object_ref = compute_on_tile(args['annotation_class_id'], args['image_id'], tile_id=args['tile_id'], sleep_time=5)
-        return {'object_ref': object_ref}, 201
-        
+        return {"tileids": [tile_id]}, 200
```

Comment:
this naming is getting a bit wild .. in this screen alone i see tileids, tile_id, and ids and i think they're all actually instances of the same information?